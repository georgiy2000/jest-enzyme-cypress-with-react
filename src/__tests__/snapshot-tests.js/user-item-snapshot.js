import React from 'react';
import renderer from 'react-test-renderer';
import UserItem from '../../components/user-item';

const fakeUser = {
  id: 1,
  name: 'Leanne Graham',
  username: 'Bret',
  address: {
    city: 'Gwenborough'
  },
  phone: '1-770-736-8031 x56442',
  company: {
    name: 'Romaguera-Crona'
  }
};

/*
 * An example of snapshot test. 
 * To see it failing, just type a letter in the first paragraph
 * of the component. 
 */
test('It renders correctly the user details', () => {
  const wrapper = renderer.create(<UserItem user={fakeUser} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
