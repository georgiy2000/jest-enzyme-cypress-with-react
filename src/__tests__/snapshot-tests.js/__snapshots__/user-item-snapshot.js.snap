// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly the user details 1`] = `
<li
  className="css-qq6j6p"
>
  <div>
    <picture>
      <source
        media="(min-width: 800px)"
        srcSet="../../images/avatar1_small.jpg"
      />
      <source
        media="(min-width: 300px)"
        srcSet="../../images/avatar1.jpg"
      />
      <img
        alt="Avatar"
        src="../../images/avatar1_small.jpg"
      />
    </picture>
    <div>
      <p>
        Leanne Graham
      </p>
      <p>
        Bret
      </p>
    </div>
  </div>
  <a
    href="/1"
    onClick={[Function]}
  >
    User Details
  </a>
</li>
`;
